### PR TITLE
Submit Assignment 3 - Processes, Tasks, & Agents

### PR DESCRIPTION
**Pull request by cogden14**
ex01.exs - Line 34

Thanks for the help!
Cole

### CODE REVIEW

**Comment by cogden14** [comment]
2 additional questions along the same lines -
    1. ex02.exs - Line 9 (& Line 50): Would you have any inclination as to why I'm receiving a 'bad return value' for the get_and_update function call?
    2. ex02.exs - Line 63: For the next value of the counter, wouldn't you expect it to be the next incremented value, i.e. 6, or would we return the value prior to incrementing as I happened to do on ex01?

**Comment by pragdave** [comment]
q1: It depends on the specification of the function: is it preincrement or post? Which way passes the tests?

part 2, q1: have a look at the warnings, and at the parameters to get_and_update

p2: to paraphrase John Lennon, whatever gets you through the tests

**Comment by cogden14** [comment]
SMU ID: 46818672

File: ex02.exs [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
-defmodule Ex02 do
 
+defmodule Ex02 do
+  @global_count {:global, __MODULE__}
+
+  def new_counter(initial_value \\ 0) do
+    { :ok, pid } = Agent.start_link(fn -> initial_value end)
+    pid
+  end
+
```
Comment (pragdave) [inline ex02.exs:10]: There's a race condition in the `next_value` function which could lead to two calls from different processes both getting the same value. Can you see where?

# -3

File: ex03.exs [hunk 2]
```diff
@@ -62,9 +62,25 @@ defmodule Ex03 do
   """
```
Comment (pragdave) [inline ex03.exs:63]: I like the way you split the functions out: it makes the pipeline easy to read.

**Comment by pragdave** [comment]
Nothing to criticize here (apart from the minor bug in ex02)

Thanks!

**Pull request closed without merging**